ec2.tf

```diff
@@ -1,26 +1,20 @@
 resource "aws_network_interface" "ilan-network-interface" {
-  count = 2
-  subnet_id   = aws_subnet.public.id
+  count           = var.deploy_count
```

Comment:
so the use of `count` is so static and repetitive... I'm thinking why do you need it in the first place? Is it just to support more than a single instance? (also remember that the names of the resources will now be appended with the index e.g. `aws_network_interface.ilan-network-interface[1]` where others do not).

Consider (and for exercise as well) creating a module for the resources and in the root repository have a `main.tf` calling your module and there it can `count` how many times to call the module.

This way the module stays "clean" from those static params and the deployment is responsible to decide how much.

Or if the reason for two instances is to have one in a private subnet and another in a public-facing subnet maybe this is better served as a Boolean variable specifying if to create a public instance or not e.g. `var.bastion_host=true`.

It may be more readable with duplicate code and duplicate resources as opposed to using `count` and trying to save a few lines of code, since terraform is declarative this is perfectly fine.